pkg/kontainer-engine/drivers/gke/gke_driver.go

```diff
@@ -554,6 +554,10 @@ func (s *state) validate() error {
 		return fmt.Errorf("minNodeCount in the NodePool must be >= 1 and <= maxNodeCount")
 	}
 
+	if s.PrivateClusterConfig != nil && s.PrivateClusterConfig.EnablePrivateEndpoint && !s.PrivateClusterConfig.EnablePrivateNodes {
```

Comment:
I thought it was the whole object, couldn't you just set `s.PrivateClusterConfig.EnablePrivateEndpoint` to false and not have nodes ? Then this check would pass but GCP would fail.

Reply:
> and not have nodes

As in, not set `EnablePrivateNodes`?

This check isn't here to fix the bug, this check is so that if a an API user (UI users are prevented from doing this) accidentally sets `EnablePrivateEndpoint` to true but `EnablePrivateNodes` to false, they are alerted to their error. Before this change, they would have been alerted by the returned error from GKE, but since now we're totally ignoring `PrivateClusterConfig` if `EnablePrivateNodes` is false, we need to do that validation on GKE's behalf.

Reply:
>  now we're totally ignoring PrivateClusterConfig if EnablePrivateNodes is false

that answers my question